src/healthchecks.js

```diff
@@ -4,13 +4,15 @@ class HealthChecks {
 	constructor(config, healthchecks) {
 		this.name = config.name;
 		this.description = config.description;
-		this.checks = config.checks
-			.filter((check) => {
-				return check.type in healthchecks;
-			})
-			.map((check) => {
-				return new healthchecks[check.type](check, this);
-			});
+		this.checks = config.checks.map((check) => {
+			if (!(check.type in healthchecks)) {
+				throw new Error(
+					`Attempted to create check '${check.name}' of type ${check.type} which does not exist`
+				);
+			}
```

Comment:
**suggestion**: instead of throwing on the first undefined check, this could collect all the undefined checks and throw an error for all of them. would prevent a user playing error whackamole

Reply:
**thought**: this would crash the app at startup right? while we'd hope that would be caught by users in local dev, there's still a chance it could creep through to review/staging apps. maybe a safer alternative would be to output a permanently-failing check for undefined checks?